backend/sftp/sftp.go

```diff
@@ -467,8 +482,36 @@ type Object struct {
 // convenience function that connects to the given network address,
 // initiates the SSH handshake, and then sets up a Client.
 func (f *Fs) dial(ctx context.Context, network, addr string, sshConfig *ssh.ClientConfig) (*ssh.Client, error) {
-	dialer := fshttp.NewDialer(ctx)
-	conn, err := dialer.Dial(network, addr)
+	var (
+		conn net.Conn
+		err  error
+	)
+	baseDialer := fshttp.NewDialer(ctx)
+	if f.opt.SocksProxy != "" {
+		var (
+			proxyAddress string
+			proxyAuth    *proxy.Auth
+		)
+		if credsAndHost := strings.SplitN(f.opt.SocksProxy, "@", 2); len(credsAndHost) == 2 {
+			proxyCreds := strings.SplitN(credsAndHost[0], ":", 2)
+			proxyAuth = &proxy.Auth{
+				User: proxyCreds[0],
+			}
+			if len(proxyCreds) == 2 {
+				proxyAuth.Password = proxyCreds[1]
+			}
+			proxyAddress = credsAndHost[1]
+		} else {
+			proxyAddress = credsAndHost[0]
+		}
+		// Build the proxy dialer
+		proxyDialer, _ := proxy.SOCKS5("tcp", proxyAddress, proxyAuth, baseDialer)
```

Comment:
Don't ignore the error please!

Reply:
I am not sure why the function returns an error. It doesn't actually trigger one.. but it looks bad so I handled it..